lib/utilities/mk-tmp-dir-in.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+var Promise   = require('../ext/promise');
+var fs        = require('fs-extra');
+var temp      = require('temp');
+var mkdir     = Promise.denodeify(fs.mkdir);
+var mkdirTemp = Promise.denodeify(temp.mkdir);
+
+function exists(dir) {
+  return new Promise(function(resolve) {
+    fs.exists(dir, resolve);
+  });
+}
+
+function mkTmpDirIn(dir) {
+  return exists(dir).then(function(doesExist) {
+    if (!doesExist) {
+      mkdir(dir);
```

Comment:
hmm, we aren't using the return value of the promise, nor does the promise actually return a value..
however, if mkdir rejects we want the containing promise to reject also. So I'll update this.